FiveCalls/FiveCalls/IssuesViewController.swift

```diff
@@ -152,70 +175,93 @@ class IssuesViewController : UITableViewController {
-        }
+        // Display 'more issues' row in the last section, if we are showing 'active' issues only.
+        let isLastSection = (viewModel.categories.count - 1 == section)
+        let moreIssuesRow = (isLastSection && !shouldShowAllIssues) ? 1 : 0
 
-        return viewModel.issues.count
+        return viewModel.categories[section].issues.count + moreIssuesRow
     }
     
     override func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
-        let notAButton = BorderedButton(frame: CGRect(x: 0, y: 0, width: tableView.frame.size.width, height: 26.0))
-        notAButton.setTitle(R.string.localizable.whatsImportantTitle(), for: .normal)
-        notAButton.setTitleColor(.fvc_darkBlueText, for: .normal)
-        notAButton.backgroundColor = .fvc_superLightGray
-        notAButton.titleLabel?.font = Appearance.instance.headerFont
-        notAButton.borderWidth = 1
-        notAButton.borderColor = .fvc_mediumGray
-        notAButton.topBorder = true
-        notAButton.bottomBorder = true
-        return notAButton
+        if shouldShowAllIssues {
+            return headerWithTitle(title: viewModel.categories[section].name)
+        }
+        return nil
     }
-    
+
     override func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
-        return 26.0
+        if shouldShowAllIssues {
+            return 26.0
+        }
+        return 0
     }
-    
+
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        guard indexPath.row < viewModel.issues.count else {
+        guard indexPath.row < viewModel.categories[indexPath.section].issues.count else {
             let cell = tableView.dequeueReusableCell(withIdentifier: R.reuseIdentifier.moreIssuesCell, for: indexPath)!
             return cell
         }
 
         let cell = tableView.dequeueReusableCell(withIdentifier: R.reuseIdentifier.issueCell, for: indexPath)!
-        let issue = viewModel.issues[indexPath.row]
+        let issue = viewModel.categories[indexPath.section].issues[indexPath.row]
+        cell.isHidden = !shouldShowAllIssues && issue.inactive
         cell.titleLabel.text = issue.name
         if let hasContacted = logs?.hasCompleted(issue: issue.id, allContacts: issue.contacts) {
             cell.checkboxView.isChecked = hasContacted
         }
         return cell
     }
 
+    override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        guard indexPath.row < viewModel.categories[indexPath.section].issues.count else {
+            return UITableViewAutomaticDimension
+        }
+        let issue = viewModel.categories[indexPath.section].issues[indexPath.row]
+        // Hide the row (height 0) if the issue for that cell is inactive and
+        // we are currently showing active issues only.
+        if !shouldShowAllIssues && issue.inactive {
+            return 0
```

Comment:
as discussed before, I really want to avoid doing this. Lean on the model to return the correct number of cells rather than faking it by making them zero height.

Reply:
I agree. But I haven't being able to figure out a way to do this, without using the main thread to do the data processing (except the current implementation and the alternate implementation I suggested in an earlier comment ie. two properties on the viewModel - activeIssues and inactiveIssues).

The requirements are:
- Show active issues on Home - don't care to which categories they belong.
- Show all issues on More - group them in categories, with categories alphabetically sorted.

ViewModel has to basically vend data for both these scenarios - so we could move some of the logic from controller to ViewModel and get rid of height==0 logic if we pre-process the data on a BG thread before handing it over to ViewModel. That way viewModel will return the correct number of cells (because it knows exactly how many activeIssues exist for a category) - removing the need for height=0 zero hack.

Reply:
This is essentially what I'm suggesting.  Take the raw list, create whatever representations you need of this data in the view model (perhaps in the background, or even one once, it's not a huge deal if on the main thread), and then just use UITableView how it was intended.

In the long run hacks like these will end up hurting us more than helping, so I think this approach will serve us better in the end.